src/download.jl

```diff
@@ -0,0 +1,41 @@
+"""
+    dataurl(dataid)
+
+This function takes in a dataset code of format "GADM_<Country Code>".\n
+Returns the URL of the gpkg dataset of the country
+"""
+function dataurl(dataid)
+    # Only accepts GADM Dataset Provider
+    if contains(dataid, "GADM")
+        provider, country_code = split(dataid, '_')
+
+        "https://biogeo.ucdavis.edu/data/gadm3.6/gpkg/gadm36_$(country_code)_gpkg.zip"
+    else
+        @error "Dataset provider not supported. Please try \"GADM_<Country Code>\"."
+    end
+end
+
+"""
+    download(dataset_code) 
+
+Registers and downloads the desired dataset.
+"""
+function download(dataset_code)
+
+    dataset_url = dataurl(dataset_code)
+
+    # This uses register function of DataDeps package
+    # Registers the dataset to be used
+    register(
+        DataDep(
+            dataset_code,
+            "GADM dataset for $dataset_code",
+            dataset_url,
+            post_fetch_method=DataDeps.unpack
+        )
+    )
+
+    # downloading the dataset if not available
+    @info "Successfully downloaded the files: ", readdir(@datadep_str dataset_code; join=true)
+    return
```

Comment:
So this return at the end is part of Julia's style guide? I find it unnecessary in most cases. If you don't mind I would just remove it.

Reply:
Julia says that functions which don't return anything need to have either `return nothing` or `return` at the end to avoid any unnecessary confusions.

Reply:
Cool. Thanks for checking that. I usually don't use return in my packages because often we have one-line functions like:

```julia
f(x) = 2x + 1
```

and a return would be too verbose. Because of that I avoid return everywhere. It is fine to keep it here as soon as you keep it consistent in the entire package :)